<?php

return array(
	array(
		'id'=>1,
		'submitter_id'=>345,
		'identifier'=>"100243",
		'title'=>'Supporting data for "Analyzing climate variations on multiple timescales can guide Zika virus response measures"',
		'description'=>'The emergence of Zika virus (ZIKV) as a public health emergency in Latin America and the Caribbean (LAC) occurred during a period of severe drought and unusually high temperatures. Speculation in the literature exists that these climate conditions were associated with the 2015/2016 El Niño event and/or climate change but to date no quantitative ',
		'dataset_size'=>1073741824,
		'ftp_site'=>'ftp://climb.genomics.cn/pub/10.5524/100001_101000/100243/',
		'upload_status'=>'Published',
		'publication_date'=>'2018-08-23',
		'publisher_id'=>1,
	),
	array(
		'id'=>2,
		'submitter_id'=>345,
		'curator_id'=>344,
		'identifier'=>"100249",
		'title'=>'Supporting data for "Analyzing climate variations on multiple timescales can guide Zika virus response measures"',
		'description'=>'The emergence of Zika virus (ZIKV) as a public health emergency in Latin America and the Caribbean (LAC) occurred during a period of severe drought and unusually high temperatures. Speculation in the literature exists that these climate conditions were associated with the 2015/2016 El Niño event and/or climate change but to date no quantitative ',
		'dataset_size'=>1073741824,
		'ftp_site'=>'ftp://climb.genomics.cn/pub/10.5524/100001_101000/100243/',
		'upload_status'=>'Published',
		'publication_date'=>'2018-08-10',
		'publisher_id'=>1,
		'is_test'=>1,
	),
	array(
		'id'=>3,
		'submitter_id'=>345,
		'curator_id'=>344,
		'identifier'=>"101001",
		'title'=>'Genome sequence of the duck (Anas platyrhynchos).',
		'description'=>'Available here is the first draft genomic sequence of the duck (Anas platyrhynchos). Duck is a member of Anatidae, a family of birds that include geese and swans. However, duck is an economically important waterfowl serving as a source of meat, eggs and feathers; though, of special interest to agriculture and medicine is that fact that the duck is a principal natural host of influenza A viruses and harbours all subtypes of 16 haemagglutinin and 9 neuraminidase subtypes currently known, except for H13 and H16 subtypes.
Using llumina Genome Analyser sequencing technology the genome of a 10-week old female Beijing duck was sequenced and a total 77 Gb of paired-end reads (approximately 64-fold coverage of the whole genome) was generated with an average length of 50 bp. Using SOAPdenovo to combine short reads, a draft genome assembly was constructed consisting of 78,487 scaffolds and covered 1.1 Gb. The contig N50 and scaffold N50 values were 26 kb and 1.2 Mb respectively. Super scaffolds were constructed and chromosomal sequences created according to the duck genetic map – this resulted in 47 superscaffolds which contained 225 scaffolds and spanned 289 Mb. Transcriptomes were also generated from several different tissues, comprising 1.87 million ESTs, and approximately 121 million 75-bp and 917 million 90-bp paired-end reads, which were generated using either the 454/Roche Life Sciences Analyzer or Illumina Genome sequencing technology. ',
		'dataset_size'=>1073741824,
		'ftp_site'=>'ftp://climb.genomics.cn/pub/10.5524/101001_102000/101001/',
		'upload_status'=>'Published',
		'publication_date'=>'2014-01-23',
		'publisher_id'=>1,
	),
	array(
		'id'=>4,
		'submitter_id'=>345,
		'curator_id'=>344,
		'identifier'=>"101000",
		'title'=>'The avian phylogenomic project data.',
		'description'=>'The evolutionary relationship of modern birds is one of the most challenging questions in systematic biology and has been debated for centuries. We proposed to rebuild the avian phylogenetic tree by using whole genome data, thus we have collected genomes of 48 bird species, representing 36 orders of bird class. The chicken, zebrafinch, and turkey genomes, which were sequenced in Sanger method, were collected from public domain. Another three genomes, pigeon, peregrine falcon, and Beijing duck, have been published during the development of this project.
The data posted here include the full genome assemblies of 45 bird species, the repeat and gene annotation produced by our new pipeline, 8295 1:1 syntenic orthologous genes, and the whole genome alignment data for all bird species. The detailed information of the published genomes can be accessed from their own publications. The genomes first released here were sequenced and assembled with NGS technology in whole genome shotgun strategy. Using an homology-based method, we annotated 13000~17000 protein-coding genes in each avian genome.
So far as we known, the avian phylogenomic project is the biggest comparative genomics project to date. The unprecedented genomic data presented here will contribute to the downstream analyses in many fields, including phylogenetics, comparative genomics, neurology, development biology, etc.
Please see the README file for a more complete description of the files that can be downloaded here.
Below are listed the links to all the individual species data used in this study. In addition, the entire dataset has been compressed into a single archive file for those who wish to retrieve the complete set.',
		'dataset_size'=>19327352832,
		'ftp_site'=>'ftp://climb.genomics.cn/pub/10.5524/100001_101000/101000/',
		'upload_status'=>'Published',
		'publication_date'=>'2014-05-16',
		'publisher_id'=>1,
	),
	array(
		'id'=>5,
		'submitter_id'=>345,
		'curator_id'=>344,
		'identifier'=>"100038",
		'title'=>'Updated genome assembly of YH: the first diploid genome sequence of a Han Chinese individual (version 2, 07/2012)',
		'description'=>'Updated genomic data from the YH (Homo sapiens) diploid genome – the first sequenced Han Chinese individual, a representative of the Asian population. The genomic DNA used in this study came from an anonymous male Han Chinese individual who has no known genetic diseases. The original version of the YH genome was assembled based on 3.3 billion reads using the Illumina Genome Analyzer (see dataset doi:10.5524/100015). This latest (as of 07/2012) and improved version of the YH genome was assembled based on 2.1 billion reads using the Illumina HiSeq2000. A total of 202G nucleotides data was achieved using 100 bp-long paired end reads with an insert size ranging from 180 bp to 40 kbp, and the genome was sequenced to 67.5-fold average coverage. The latest version of SOAPdenovo2 was used to reassemble, improve and update the previously assembled genome (tools and pipelines available here: doi:10.5524/100044). By aligning the short reads with SOAP, 177G nucleotides were mapped onto the NCBI reference genome and 99.99% of the genome was covered. The raw sequences, assemblies and relevant tools are released for public use under a CC0 license. More information about the YH genome can be viewed at: http://yh.genomics.org.cn/',
		'dataset_size'=>83751862272,
		'ftp_site'=>'ftp://climb.genomics.cn/pub/10.5524/100001_101000/100038/',
		'upload_status'=>'Published',
		'publication_date'=>'2012-12-12',
		'publisher_id'=>1,
	),
	array(
		'id'=>6,
		'submitter_id'=>345,
		'curator_id'=>344,
		'identifier'=>"100044",
		'title'=>'Software and supporting material for “SOAPdenovo2: An empirically improved memory-efficient short read de novo assembly”',
		'description'=>'SOAPdenovo2 is the latest de novo genome assembly package from BGI’s SOAP (short oligonucleotide analysis package) suite of tools (homepage here: http://soap.genomics.org.cn/). Compared to SOAPdenovo1, this new version has the advantage of a new algorithm design that reduces memory consumption in graph construction, resolves more repeat regions in contig assembly, increases coverage and length in scaffold construction, improves gap closure, and is optimized for large genomes.
Using new sequencing data from the YH (Homo sapiens) diploid genome – the first sequenced Han Chinese individual, an updated assembly was produced (see dataset here: doi:10.5524/100038), with the N50 scores for the contig and scaffold being 3-fold and 50-fold longer, respectively, than the first published version. The genome coverage increased from 81.16% to 93.91%, and memory consumption was ~2/3 times lower during the point of largest memory consumption.
Benchmarking with Assemblathon1 and GAGE datasets shows that SOAPdenovo2 greatly surpasses its predecessor SOAPdenovo1 and is competitive to other assemblers on both assembly length and accuracy.
In order to facilitate readers to repeat and recreate these findings, configured packages with the compressed pipelines containing all of the necessary shell scripts and tools are available from the BGI FTP server (ftp://public.genomics.org.cn/BGI/SOAPdenovo2).
The latest version of SOAPdenovo2 is available from Sourceforge: http://soapdenovo2.sourceforge.net/
These pipelines are available from our data platform as Galaxy workflows: http://galaxy.cbiit.cuhk.edu.hk/',
		'dataset_size'=>268435456000,
		'ftp_site'=>'ftp://climb.genomics.cn/pub/10.5524/100001_101000/100044/',
		'upload_status'=>'Published',
		'publication_date'=>'2012-12-13',
		'publisher_id'=>1,
	),
	array(
		'id'=>7,
		'submitter_id'=>345,
		'curator_id'=>344,
		'identifier'=>"100148",
		'title'=>'Supporting materials for "SOAPdenovo2 Case Study".',
		'description'=>'The SOAPdenovo2 case study is a reproducibility study aimed at exploring how existing research objects and workflow enactment engines can help assess, record and preserve scientific workflows and associated findings by reviewing a comparison between sequence assembly algorithm performance in the light of development of the SOAPdenovo2 de novo genome assembler. The case study was a joint effort by the GigaScience journal, the Investigation/Study/Assay (ISA) infrastructure, Nanopublication (Nanopub) and Research Object (RO) communities and SOAPdenovo2 de novo genome assembler developers. Included here are the relevant ISA-TAB, linkedISA, nanopubs and Research Object files. For more see the paper and GitHub page.',
		'dataset_size'=>1048576,
		'ftp_site'=>'ftp://climb.genomics.cn/pub/10.5524/100001_101000/100148/',
		'upload_status'=>'Published',
		'publication_date'=>'2015-06-09',
		'publisher_id'=>1,
	),
);