<?php
$this->pageTitle='GigaDB - About';

//echo $this->renderInternal('Yii::app()->basePath'.'/../files/html/about.html');
?>

<div class="clear"></div>
<div class="row">
    <div class="about-tabs-container">
        <a class="btn about-tabs" href="/site/about">General Information</a>
        <a class="btn about-tabs" href="/site/advisory">Advisory Panel</a>
        <a class="btn about-tabs tab-active" href="/site/faq">FAQ</a>
    </div>
    <div class="about1">        
        <h4>Contents</h4>
<ol >
<li >
  <a href="#TOC-What-is-GigaDB-">
  What is GigaDB?
  </a>
</li>
<li >
  <a href="#TOC-What-journals-are-integrated-with-GigaDB-">
  What journals are integrated with GigaDB ?
  </a>
</li>
<li >
  <a href="#TOC-Why-use-GigaDB-">
  Why use GigaDB?
  </a>
</li>
<li ><a href="#TOC-What-kinds-of-data-does-GigaDB-accept-">What kinds of data does GigaDB accept?</a></li>
<li ><a href="#TOC-My-research-is-on-human-subjects.-Can-I-archive-my-data-in-GigaDB-">My research is on human subjects. Can I archive my data in GigaDB?</a></li>
<li ><a href="#TOC-In-what-file-format-s-should-I-submit-my-data-">In what file format(s) should I submit my data?</a></li>
<li ><a href="#TOC-When-should-I-submit-my-data-">When should I submit my data?</a></li>
<li ><a href="#TOC-How-can-I-modify-files-I-have-submitted-to-GigaDB-while-my-article-is-in-review-">How can I modify files I have submitted to GigaDB while my article is in review?</a></li>
<li ><a href="#TOC-What-should-I-prepare-before-submission-">What should I prepare before submission?</a></li>
<li ><a href="#TOC-How-can-I-make-my-data-submission-as-accessible-and-reusable-as-possible-">How can I make my data submission as accessible and reusable as possible?</a></li>
<li ><a href="#TOC-How-do-I-submit-data-">How do I submit data?</a></li>
<li ><a href="#TOC-How-do-I-write-a-ReadMe-file-">How do I write a ReadMe file?</a></li>
<li ><a href="#TOC-How-do-I-cite-the-data-in-my-manuscript-">How do I cite the data in my manuscript?</a></li>
<li ><a href="#TOC-Are-there-any-problems-with-publishing-my-final-research-paper-AFTER-publishing-the-data-in-GigaScience-">Are there any problems with publishing my final research paper AFTER publishing the data in GigaScience?</a></li>
<li ><a href="#TOC-Do-I-have-the-option-to-embargo-release-of-my-data-">Do I have the option to embargo release of my data?</a></li>
<li ><a href="#TOC-How-much-does-it-cost-">How much does it cost?</a></li>
<li ><a href="#TOC-Do-I-have-to-pay-to-download-or-use-the-data-">Do I have to pay to download or use the data?</a></li>
<li ><a href="#TOC-How-do-I-download-a-large-dataset-with-my-slow-internet-connection-">How do I download a large dataset with my slow internet connection ?</a></li>
<li ><a href="#TOC-How-do-I-cite-data-from-GigaDB-">How do I cite data from GigaDB ?</a></li>
<li ><a href="#TOC-How-do-I-download-information-to-my-citation-management-software-">How do I download information to my citation management software?</a></li>
<li ><a href="#TOC-What-is-a-dataset-">What is a dataset?</a></li>
<li ><a href="#TOC-Does-my-journal-work-with-GigaDB-and-how-">Does my journal work with GigaDB and how?</a></li>
<li ><a href="#TOC-What-is-a-GigaDB-DOI-">What is a GigaDB DOI?</a></li>
<li ><a href="#TOC-Why-does-GigaDB-use-Creative-Commons-Zero-">Why does GigaDB use Creative Commons Zero?</a></li>
<li ><a href="#TOC-Can-the-GigaDB-repository-help-me-prepare-a-data-management-plan-">Can the GigaDB repository help me prepare a data management plan?</a></li>
<li ><a href="#TOC-What-are-the-charges-for-submitting-data-">What are the charges for submitting data?</a></li>
<li ><a href="#TOC-Why-is-submission-to-GigaDB-not-closely-integrated-with-submission-to-GigaScience-">Why is submission to GigaDB not closely integrated with submission to GigaScience?</a></li>
<li ><a href="#TOC-How-are-datasets-in-GigaDB-backed-up-">How are datasets in GigaDB backed up?</a></li>
<li ><a href="#TOC-What-happens-to-data-after-it-is-submitted-">What happens to data after it is submitted?</a></li>
<li ><a href="#TOC-Can-I-see-how-often-my-dataset-is-being-used-and-downloaded-">Can I see how often my dataset is being used and downloaded?</a></li>
<li ><a href="#TOC-How-may-data-from-GigaDB-be-reused-">How may data from GigaDB be reused?</a></li>
<li ><a href="#TOC-What-is-the-side-bar-on-the-right-of-all-dataset-pages-">What is the side-bar on the right of all dataset pages?</a></li>
<li ><a href="#TOC-What-is-Hypothes.is-">What is Hypothes.is ? </a></li>
<li ><a href="#TOC-How-do-I-report-missing-values-in-my-metadata-">How do I report missing values in my metadata? </a></li>

</ol>

<br />
<br />

<h4><a name="TOC-What-is-GigaDB-"></a>What is GigaDB?</h4>

GigaDB is the home for all data/files/tools/software associated with <i>GigaScience </i>manuscripts. GigaDB curators will ensure the information is complete and appropriately formatted, before cataloging and publishing. Submission of data to GigaDB complements but does not serve as a replacement for community approved public repositories, supporting data and source code should still be made publicly available in a suitable public repository. GigaDB can link any and all publicly deposited data together with additional files/tools that do not have a natural home in any other public repository. 

<br />
<br />

<h4><a name="TOC-What-journals-are-integrated-with-GigaDB-"></a>What journals are integrated with GigaDB?</h4>
At the present only <i>GigaScience</i>.

<br />
<br />

<h4><a name="TOC-Why-use-GigaDB-"></a>Why use GigaDB?</h4>
<i>GigaScience</i> is committed to enabling reproducible science, to do this, readers need to be able to easily find and get hold off all the underlying data, methods, workflows, software and anything else that was used in the research. In the past authors of research articles have made (justified) claims that there is no way of making all their data available, now GigaDB has filled that gap. GigaDB complements but does not serve as a replacement for community approved public repositories, and can link any and all publicly deposited data together with additional files/tools that do not have a natural home in any other public repository. Any and all data/files required for reproducibility of <i >GigaScience </i>manuscripts should be either hosted in, or linked to from, GigaDB.

<br />
<br />

<h4><a name="TOC-What-kinds-of-data-does-GigaDB-accept-"></a>What kinds of data does GigaDB accept?</h4>
Anything related to a <i>GigaScience </i>article that does NOT already have a relevant public repository (e.g. sequence data should still be deposited in the INSDC archives and/or the SRA).

<br />
<br />

<h4><a name="TOC-My-research-is-on-human-subjects.-Can-I-archive-my-data-in-GigaDB-"></a>My research is on human subjects. Can I archive my data in GigaDB?</h4>
As long as the data is fully consented and legally and ethically approved for public release, we encourage complete disclosure including a blank copy of the consent form signed where possible.

<br />
<br />

<h4><a name="TOC-In-what-file-format-s-should-I-submit-my-data-"></a>In what file format(s) should I submit my data?</h4>
There are many specific formats depending on the date types, but the rule of thumb is to use non-propitiatory formats and where possible follow the standard for the relevant field, our curators will be on hand to help with any specific questions on this matter.

<br />
<br />

<h4><a name="TOC-When-should-I-submit-my-data-"></a>When should I submit my data?</h4>At the same time or soon after submission of the manuscript. <i>GigaScience </i>reviewers will be looking to see if the underlying data is available and appropriate, so they will need access to it. This can be via your own private servers if you prefer, but we offer a secure staging area to host data under review/pre-release.

<br />
<br />

<h4><a name="TOC-How-can-I-modify-files-I-have-submitted-to-GigaDB-while-my-article-is-in-review-"></a>How can I modify files I have submitted to GigaDB while my article is in review?</h4>While the dataset is "pre-release" status, any files can be replaced by over-writing the original file with the newly modified one. After publishing the data, no over-writing will be permissible, only the addition of new files, all published files will remain available (unless there is a very good reason to remove them). Versioning is still possible for updates and major changes to the files if they need to be changed post-publication.

<br />
<br />

<h4><a name="TOC-What-should-I-prepare-before-submission-"></a>What should I prepare before submission?</h4>Organise your data into logical directories/folders, name files consistently and <b>without using spaces or special characters</b>. Re-read you methods section to check that everything mentioned there is available, either by links to public repositories or as files you have organised to submit to GigaDB.

<br />
<br />

<h4><a name="TOC-How-can-I-make-my-data-submission-as-accessible-and-reusable-as-possible-"></a>How can I make my data submission as accessible and reusable as possible?</h4>By ensuring all files are in non-proprietary formats that anyone can use without the need for expensive software.By making sure data tables are in tables <i>not PDF</i>.By using a CC0 waiver or other suitable public domain licenses for datasets. By using OSI (Open Source Initiative) licenses for software, and linking to versions in code repositories for updates and forking.By including as much metadata about the samples/specimens/files/methods etc... as possible.

<br />
<br />

<h4><a name="TOC-How-do-I-submit-data-"></a>How do I submit data?</h4>All data submissions should be approved before being started, please contact <a href="mailto:editorial@gigasciencejournal.com">editorial@gigasciencejournal.com</a> to discuss your article and associated data with our editors.Once approved, there are two possible routes to provide the metadata about your data:
<ol><li> use the online <a href="http://gigadb.org/dataset/upload">submission wizard</a> - this is a good option for datasets with few authors, and few files. The wizard currently does not have functionality to upload tabular information so everything must be typed in individually.</li>
<li> use the template spreadsheet (excel, but compatible with open office too) downloadable from <a href="http://www.google.com/url?q=http%3A%2F%2Fgigadb.org%2Ffiles%2FGigaDBUploadForm.xls&amp;sa=D&amp;sntz=1&amp;usg=AFrqEzf6cPyp-vGxxVTE9OEq2TPsJqU7fg">here </a>- This option is better where there are multiple authors and/or multiple files and/or samples. NB. the spreadsheet contains macros, but these are only to allow the forward and back buttons to work so can be disabled, you can just click the relevant tabs at the bottom of the spreadsheet.
</li></ol>
For more details on submitting using the Spreadsheet please see here.

<br />
<br />

<h4><a name="TOC-How-do-I-write-a-ReadMe-file-"></a>How do I write a ReadMe file?</h4>The readme file is an important part of any dataset and our curators will be able and willing to assist with this if required. We intend to formalise the readme format at some point in the near future, but for now here is an example of the format we try to work to:
<br />
<br />
filename = readme.txt
<br />format = ASCII plain text (<b>not RTF, not .doc</b> !)
<br />
<br />
&lt;Dataset title&gt;
<br />
==========
<br />
&lt;Author list&gt;:&lt;year&gt;, GigaScience database, &lt;DOI&gt;
<br />
summary:
<br />
---------
</br />
[optionally you may include a summary text about the dataset or directory structure used here]
<br />
Associated data:
<br />
--------------
<br />
[list any URL links or DOIs to other public repository data]
<br />
<br />
Directories:
<br />
----------
<br />
[list any directories of related files with a description to help users understand why these files are grouped into a directory]
<br />
&lt;directory_name&gt; - &lt;description of the group of files in the directory&gt;
<br />
<br />
Files:
<br />
-----
<br />
[list the files available in this dataset with a brief description for each]
<br />
&lt;filename&gt; - &lt;description&gt;

<br />
<br />

<h4><a name="TOC-How-do-I-cite-the-data-in-my-manuscript-"></a>How do I cite the data in my manuscript?</h4><i>GigaScience</i> supports and has signed the <a href="https://www.force11.org/group/joint-declaration-data-citation-principles-final" target="_blank">FORCE 11 Joint Declaration of the Data Citation Principles</a>, feeling strongly that data should be accorded the same importance in the scholarly record as citations of publications. GigaDB datasets can and should be cited in the same manner as any other reference, although the format is journal specific based on their instructions. Following DCC and DataCite guildelines, in GigaScience journal the citation within the references section will be of the form:Author List; publication year: "Dataset title", GigaScience Database. DOI. Example.Peter E Larsen; Yang Dai; (2015): Supporting materials for "Metabolome of Human gut microbiome is predictive of host dysbiosis".; GigaScience Database. http://dx.doi.org/10.5524/100163

<br />
<br />

<h4><a name="TOC-Are-there-any-problems-with-publishing-my-final-research-paper-AFTER-publishing-the-data-in-GigaScience-"></a>Are there any problems with publishing my final research paper AFTER publishing the data in <i>GigaScience</i>?</h4>There shouldn't be, and a major rationale for data publishing is to incentivise earlier release of data in this manner. It is commonly understood throughout the publishing community that publishing data (as a Data Note or in a public archive) is a good thing to be encouraged, and as such, there are no penalties to then subsequently publishing research based on those data. <i >GigaScience</i> has published plenty of data notes and released data sets prior to the analysis papers being published, some examples are: 

<ul>
<li>3,000 Rice Genomes Project (13.4 Tb data).
</li>
<li>Polar Bear genome - dataset released in GigaDB nearly 3 years before the analysis paper was published in <i>Cell</i>.
</li>
<li>Deadly 2011 outbreak <a href="http://dx.doi.org/10.5524/100001" target="_blank"><i>E. Coli</i> genome</a> that lead to over 50 deaths in Germany (and eventually published in <i>NEJM</i>).
</li>
</ul>

<p >Our Polar bear genome data was released nearly three years before any official publication came out from the project, and despite being used by at least 5 other studies, the analysis paper made the cover of <i>Cell</i> (see <a href="http://blogs.biomedcentral.com/gigablog/2014/05/14/the-latest-weapon-in-publishing-data-the-polar-bear/" target="_blank">the blog</a> for more). </p>
<p >Journals do not consider the publication of a dataset with a DOI and associated protocol information as a 'prior publication' that would preclude subsequent publication of new results obtained from such a dataset. F1000 Research did a useful survey to confirm this with a number of publishers (see: <a href="http://f1000research.com/data-policies" target="_blank"> F1000 policy</a>), and this is only going to become increasingly observed and accepted as most of the publishers are now promoting their own Data Journals.</p>

<br />
<br />

<h4><a name="TOC-Do-I-have-the-option-to-embargo-release-of-my-data-"></a>Do I have the option to embargo release of my data?</h4>As early release as possible is encouraged, although the standard protocol we follow is to maintain data as private to the peer reviewers only until the associated manuscript has been formally accepted, at which point the dataset is released, this is usually several days prior to the manuscript publication due to production times of the BMC publishing system. While we cannot foresee any reason why datasets should be embargoed for extended periods we can discuss this further on a case-by-case basis.If you have major concerns about someone else publishing on your data before you, we can add a Fair Use policy statement on the GigaDB dataset page which looks like this:
<br />
<img alt="policy" src="http://gigadb.org/images/fair_use2.gif"  /><p >These data are made available pre-publication under the recommendations of the <a href="https://www.genome.gov/10506537" target="_blank">Fort Lauderdale</a>/<a href="http://www.nature.com/nature/journal/v461/n7261/full/461168a.html">Toronto</a> meetings. Please respect the rights of the data producers to publish their whole dataset analysis first. The data is being made available so that the research community can make use of them for more focused studies without having to wait for publication of the whole dataset analysis paper. If you wish to perform analyses on this complete dataset, please contact the authors directly so that you can work in collaboration rather than in competition.</p><p ><b>This dataset fair use agreement is in place until </b><b ><i>&lt;author can specify a data up to 12 months away&gt;</i></b></p><p ></p>

<br />
<br />

<h4><a name="TOC-How-much-does-it-cost-"></a>How much does it cost?</h4>There are currently no separate <u >d</u>ata <u >p</u>ublishing <u >c</u>harges (DPC's) for GigaDB as we currently do not accept data that is not accompanied by a <i >GigaScience </i>manuscript. All DPCs for <i >GigaScience </i>manuscripts are covered by the <u >a</u>rticle <u >p</u>ublishing <u >c</u>harges (APC's) of that manuscript (up to a terabyte automatically included, but contact us if you need more). For APCs of <i >GigaScience </i>manuscripts please see <a href="http://www.gigasciencejournal.com/about/apcfaq/howmuch" >here</a>.

<br />
<br />

<h4><a name="TOC-Do-I-have-to-pay-to-download-or-use-the-data-"></a>Do I have to pay to download or use the data?</h4>No. All data provided by GigaDB is free to download and use. On occasion when datasets are very large and internet connections are slow, some user may request data to be sent by hard disk, GigaDB cannot bare the cost of this but we will assist in the copy of the data onto the disks and help arrange shipment, but the user will be required to cover the cost of the disks and shipment.

<br />
<br />

<h4><a name="TOC-How-do-I-download-a-large-dataset-with-my-slow-internet-connection-"></a>How do I download a large dataset with my slow internet connection ?</h4>There are 2 ways to download data from GigaDB:
<ol>
<li> FTP. This is the "normal" method, click the download button on any dataset page and this is how your data will be sent.
</li>
<li> Hard drive shipment. On occasion when datasets are very large and internet connections are slow, some user may request data to be sent by hard disk, GigaDB cannot bare the cost of this but we will assist in the copy of the data onto the disks and help arrange shipment, but the user will be required to cover the cost of the disks and shipment.
</li>
</ol>
<br />
<br />

<h4><a name="TOC-How-do-I-cite-data-from-GigaDB-"></a>How do I cite data from GigaDB ?</h4><i>GigaScience</i> supports and has signed the <a href="https://www.force11.org/group/joint-declaration-data-citation-principles-final" target="_blank">FORCE 11 Joint Declaration of the Data Citation Principles</a>,
 feeling strongly that data should be accorded the same importance in the scholarly record as citations of publications. GigaDB datasets can and should be cited in the same manner as any other reference, although the format is journal specific based on their instructions. Following DCC and DataCite guildelines, in GigaScience journal the citation within the references section will be of the form:Author List; publication year: "Dataset title", GigaScience Database. DOI. Example.Peter E Larsen; Yang Dai; (2015): Supporting materials for "Metabolome of Human gut microbiome is predictive of host dysbiosis".; GigaScience Database. http://dx.doi.org/10.5524/100163

<br />
<br />

<h4><a name="TOC-How-do-I-download-information-to-my-citation-management-software-"></a>How do I download information to my citation management software?</h4>On each dataset page there are 3 buttons after the authors names, "RIS", "BIBTEX" and "TEXT" you may use these to download the citation of the dataset in those formats.

<br />
<br />

<h4><a name="TOC-What-is-a-dataset-"></a>What is a dataset?</h4>The term dataset in GigaDB refers to a collection of related works, including but not limited to; files, software, workflows, experiments, data, metadata and results. Each dataset has its own webpage which has a DOI (digital object identifier). These datasets are permanent and citable records of research output designed to allow for a modernization of the classical publishing framework while maintaining the familiarity of citations and metrics thereof.While uncommon, it is possible for a dataset to be made-up of several other datasets in a nested fashion, for example the Avian phylogenomics project data dataset (<a href="http://dx.doi.org/10.5524/101000">http://dx.doi.org/10.5524/101000</a>) is a compilation of 48 other datasets, some of those were published before and some at the same time. This allows the original authors to cite just one dataset to cover them all, but also allows future users to cite individual datasets if they require. We will discuss the merits of such procedures on a case-by-case basis with the submitter.

<br />
<br />

<h4><a name="TOC-Does-my-journal-work-with-GigaDB-and-how-"></a>Does my journal work with GigaDB and how?</h4>While we have no formal agreements with any particular journals, we are happy to work with other journals to ensure timely and coherent joint publications, please discuss with the editors (<a href="mailto:editorial@gigasciencejournal.com">editorial@gigasciencejournal.com</a>)

<br />
<br />

<h4><a name="TOC-What-is-a-GigaDB-DOI-"></a>What is a GigaDB DOI?</h4>A Digial Object Identifier (DOI) is a stable, citable link to an electronic resource. A GigaDB DOI is a stable and citable link to a dataset hosted by GigaDB. 

<br />
<br />

<h4><a name="TOC-Why-does-GigaDB-use-Creative-Commons-Zero-"></a>Why does GigaDB use Creative Commons Zero?</h4>It is widely recognized that publicly funded research data should be made publicly available for free to be used by anyone. The Creative Commons Zero (CC0) waiver provides the explicit statement of that fact, and it is transparent to all that the data hosted by GigaDB are all freely available for any use case. CC0 is thought to be the most appropriate method for dedicating data to the public domain, but for more on the rationale and practicalities <a href="http://www.biomedcentral.com/1756-0500/5/494" target="_blank">see this <i>BMC Research Notes</i> editorial</a>. Citation of data usage is greatly encouraged in order to provide the recognition to the data producers, both for their efforts in the production and in their foresight and generosity in making the data CC0.

<br />
<br />

<h4><a name="TOC-Can-the-GigaDB-repository-help-me-prepare-a-data-management-plan-"></a>Can the GigaDB repository help me prepare a data management plan?</h4>At the present time GigaDB doesn't have the resources to assist in data management plans, but there are many useful resources available on the internet, including places like the <a href="http://www.dcc.ac.uk/resources/data-management-plans">DCC (UK focus)</a>or <a href="https://www.cic.net/projects/technology/shared-storage-services/data-management-plans">the CIC (US focus)</a>

<br />
<br />

<h4><a name="TOC-What-are-the-charges-for-submitting-data-"></a>What are the charges for submitting data?</h4>There are currently no separate <u>d</u>ata <u>p</u>ublishing <u>c</u>harges (DPC's) for GigaDB as we currently do not accept data that is not accompanied by a <i>GigaScience </i>manuscript. All DPCs for <i>GigaScience </i>manuscripts are covered by the <u>a</u>rticle <u>p</u>ublishing <u>c</u>harges (APC's) of that manuscript (up to a terabyte automatically included, but contact us if you need more). For APCs of <i>GigaScience </i>manuscripts please see <a href="http://www.gigasciencejournal.com/about/apcfaq/howmuch">here</a>.

<br />
<br />

<h4><a name="TOC-Why-is-submission-to-GigaDB-not-closely-integrated-with-submission-to-GigaScience-"></a>Why is submission to GigaDB not closely integrated with submission to <i>GigaScience</i>?</h4>Due to various differences in the BMC's editorial tools and the GigaDB system, unfortunately at this time it is not possible to integrate the submission process, but our editors and curators will do everything they can to make the process as smooth as possible for authors.

<br />
<br />

<h4><a name="TOC-How-are-datasets-in-GigaDB-backed-up-"></a>How are datasets in GigaDB backed up?</h4>We have a regular backup of data, so if you find a corrupt file please let us know and we will replace it with a copy from back-up.

<br />
<br />

<h4><a name="TOC-What-happens-to-data-after-it-is-submitted-"></a>What happens to data after it is submitted?</h4>We will host your data on our private GigaDB server giving access to the reviewers, if the manuscript is accepted we will move the data to our public production server. If it is unsuccessful the data will be deleted.

<br />
<br />

<h4><a name="TOC-Can-I-see-how-often-my-dataset-is-being-used-and-downloaded-"></a>Can I see how often my dataset is being used and downloaded?</h4>urrrm. Sort-of yes. If a user clicks the download button on the website, it is recorded in the database and you can see on the dataset page how many times this has happened. However, if a file is pulled directly from the FTP server it is currently not recorded in the database. This functionality is on our to-do list and will be addressed as soon as we can.

<br />
<br />

<h4><a name="TOC-How-may-data-from-GigaDB-be-reused-"></a>How may data from GigaDB be reused?</h4>It can be used for anything by anyone, most* data is given the licence CC0 specifically to remove any restrictions on reuse. * - on occasion we host some files for convenience of our users that are already covered by other licences (e.g. more appropriate OSI-compliant licenses for software, or multiple (all open) licenses in a workflow or virtual machine), where this happens we make every effort to make users aware of the different licences.

<br />
<br />

<h4><a name="TOC-What-is-the-side-bar-on-the-right-of-all-dataset-pages-"></a>What is the side-bar on the right of all dataset pages?</h4><p>This is the hypothes.is annotation tool bar. See what is Hypothes.is for more details.</p>

<br />
<br />

<h4><a name="TOC-What-is-Hypothes.is-"></a>What is Hypothes.is ? </h4><h4><a name="TOC-1"></a><a href="http://hypothes.is/" imageanchor="1"  target="_blank"><img alt="http://hypothes.is" border="0" src="https://sites.google.com/a/gigasciencejournal.com/gigascience/_/rsrc/1447915702699/GigaDB/faqs/hypothesis.png"  /></a></h4><a href="http://gigadb.org/images/hypothesis.png">Hypothes.is</a> is an open source project helping to bring a discussion, annotation and curation layer to the web, we are collaborating with Hypothes.is in order to make all our datasets open to discussion by anyone (with a hypothesis user account). Simply highlight the text of interest on the page and click "New Note" icon that appears. To see previous notes, click the number on the side bar, or open the side bar to see all preious public annotations.

<br />
<br />

<h4><a name="TOC-How-do-I-report-missing-values-in-my-metadata-"></a>How do I report missing values in my metadata?</h4><p>A. There are various reason why certain data values may need to not be included in the sample metadata, but you still want it to be compliant with particular Minimum Information standards such the GSC MIxS. To maintain compliance when there are missing values within the mandatory fields please use the following terms only:<br>
 Term - Definition<br>
 <b>not applicable</b> - information is inappropriate to report, can indicate that the standard itself fails to model or represent the information appropriately<br>
 <b>restricted access</b> - information exists but can not be released openly because of privacy concerns<br>
 <b>not provided</b> - information is not available at the time of submission, a value may be provided at the later stage<br>
 <b>not collected</b> - information was not collected and will therefore never be available</p>
 
<br />
<br />  
    </div>
</div>

